### DISCUSSION TITLE
the multiple imputation in mice()  for ordinal data using logistic regression

### DISCUSSION THREAD

**Discussion by AJasz** [amices/mice#390] [Q&A]
Hi @stefvanbuuren ,

These are the code i tried to implement the multiple imputation in a set of questionnaire.
All the data types are ordered ordinal from 1 to 5 in which the 1 represents Strongly disagree while 5 represents Strongly agree.

These are the codes:
library(mice)

dat <- read.csv(file = 'data inhumanity122620-2.csv')
datOri<-dat[,c(1,4:22)]
dat<-datOri

dat$Perspective<-factor(dat$Perspective)
dat$Gender<-factor(dat$Gender)
dat$Em1<-factor(dat$Em1)
dat$Em2<-factor(dat$Em2)
dat$Em3<-factor(dat$Em3)

dat$SN1<-factor(dat$SN1)
dat$SN2<-factor(dat$SN2)
dat$SN3<-factor(dat$SN3)

dat$Ec1<-factor(dat$Ec1)
dat$Ec2<-factor(dat$Ec2)
dat$Ec3<-factor(dat$Ec3)

summary(dat)

female <- dat$Emotion6 == 1
NaFm05 <- sum(female) * 0.05
NaFm10 <- sum(female) * 0.1
NaFm15 <- sum(female) * 0.15
NaFm20 <- sum(female) * 0.2

dat$Em3[sample(which(female),NaFm05)]<-NA

datImpute<-dat[,-1]
impMAR05 =mice(datImpute, m=5, printFlag=FALSE, maxit = 30, seed=2525,meth='polr')

model_impMAR05 <- with(impMAR05, glm(Perspective ~ Em1 + Em2 + Em3 +
                                       SN1+SN2+SN3+Ec1+Ec2+Ec3,
                                       family = binomial(link = 'logit' )))

tab051 <- summary(pool(model_impMAR05), conf.int = TRUE, exponentiate = TRUE)

It is resulting these:

Warning messages:
1: In qt(a, df) : NaNs produced
2: In qt(a[1], df[parm]) : NaNs produced
3: In qt(a[2], df[parm]) : NaNs produced
> tab051
          term     estimate std.error     statistic df   p.value 2.5 % 97.5 %
1  (Intercept) 1.268244e+11  648000.3  3.945379e-05  0 0.9999988   NaN    NaN
2         Em12 1.608434e+22  916410.7  5.579609e-05  0 0.9999982   NaN    NaN
3         Em13 9.999957e-01  965982.0 -4.476533e-12  0 1.0000000   NaN    NaN
4         Em14 9.999956e-01  610940.7 -7.255558e-12  0 1.0000000   NaN    NaN
5         Em22 9.999978e-01  432000.0 -5.118755e-12  0 1.0000000   NaN    NaN
6         Em23 1.000011e+00 1331514.2  8.306376e-12  0 1.0000000   NaN    NaN
7         Em24           NA        NA            NA NA        NA    NA     NA
8         Em25 1.000011e+00 1432782.8  7.719152e-12  0 1.0000000   NaN    NaN
9         Em32 9.999955e-01  432000.2 -1.046423e-11  0 1.0000000   NaN    NaN
10        Em33 9.999910e-01 1013130.4 -8.835569e-12  0 1.0000000   NaN    NaN
11        Em34 1.000004e+00  683052.3  6.470151e-12  0 1.0000000   NaN    NaN
12        Em35 9.999976e-01 1058180.0 -2.284788e-12  0 1.0000000   NaN    NaN
13        SN12 9.999979e-01  610940.6 -3.465096e-12  0 1.0000000   NaN    NaN
14        SN13 1.000002e+00  305470.3  7.568614e-12  0 1.0000000   NaN    NaN
15        SN14 6.217199e-23  683052.3 -7.485831e-05  0 0.9999976   NaN    NaN
16        SN15 1.000005e+00  808198.4  5.718703e-12  0 1.0000000   NaN    NaN
17        SN22 9.999979e-01  683052.4 -3.094147e-12  0 1.0000000   NaN    NaN
18        SN23 9.999978e-01  610940.6 -3.633881e-12  0 1.0000000   NaN    NaN
19        SN24 9.999957e-01  683052.3 -6.332929e-12  0 1.0000000   NaN    NaN
20        SN25 1.000002e+00  808198.4  2.976443e-12  0 1.0000000   NaN    NaN
21        SN32 1.000004e+00  683052.3  6.332929e-12  0 1.0000000   NaN    NaN
22        SN33 1.000002e+00  305470.2  7.251873e-12  0 1.0000000   NaN    NaN
23        SN34 1.000002e+00  305470.2  7.243278e-12  0 1.0000000   NaN    NaN
24        SN35 9.999977e-01  305470.3 -7.572242e-12  0 1.0000000   NaN    NaN
25        Ec12           NA        NA            NA NA        NA    NA     NA
26        Ec13           NA        NA            NA NA        NA    NA     NA
27        Ec14           NA        NA            NA NA        NA    NA     NA
28        Ec15           NA        NA            NA NA        NA    NA     NA
29        Ec22           NA        NA            NA NA        NA    NA     NA
30        Ec23           NA        NA            NA NA        NA    NA     NA
31        Ec24           NA        NA            NA NA        NA    NA     NA
32        Ec32           NA        NA            NA NA        NA    NA     NA
33        Ec33           NA        NA            NA NA        NA    NA     NA
34        Ec34           NA        NA            NA NA        NA    NA     NA
35        Ec35           NA        NA            NA NA        NA    NA     NA

In the result, Em starts at 12, but the actual variable starts at Em1. I do confused why should NAs exist in the result?
Could you please advice, how to write the code to apply the mi in survey?
Thanks in advance

**Comment by stefvanbuuren** [comment]
Don't have time to dive in. Does not seem to be a software issue, so transferring to discussions.
